moz_logging/sandboxes/heka/analysis/moz_logging_ingestion_error_monitor.lua

```diff
@@ -160,13 +168,49 @@ Diagnostic (count/error)
 ========================
 %s
 ]]
+local function error_alert(ns, th, k, cb, s, e)
+    local array = cb:get_range(INGESTED, s, e)
+    local isum, vcnt = stats.sum(array)
+    array = cb:get_range(ERROR, s, e)
+    local esum = stats.sum(array)
+    if isum > 1000 or esum > 1000 then
+        local pe  = esum / (isum + esum) * 100
+        if pe > th.percent then
+            if alert.send(k, "ingestion error",
+                          string.format(ingestion_error_template, isum, esum, pe,
+                                        th.percent, alert.get_dashboard_uri(k),
+                                        diagnostic_dump(diags))) then
+                cb:annotate(ns, ERROR, "alert", string.format("%.4g%%", pe))
+            end
+        end
+    end
+    return vcnt
+end
+
+
+local function inactivity_alert(ns, th, k, vcnt, cb, e)
+    local iato = th.inactivity_timeout
+    if vcnt == 0 or iato == 0 then return end
+
+    if MINS_IN_HOUR - vcnt > iato then
+        local _, cnt = stats.sum(cb:get_range(INGESTED, e - ((iato - 1) * 60e9))) -- include the current minute
+        if cnt == 0 then
+            if alert.send(k, "inactivitiy timeout",
+                          string.format("No new valid data has been seen in %d minutes\n\ngraph: %s\n",
+                                        iato, alert.get_dashboard_uri(k))) then
+                cb:annotate(ns, INGESTED, "alert", "inactivitiy timeout")
+            end
+        end
+    end
+end
+
 
 function timer_event(ns, shutdown)
     for k, v in pairs(inputs) do
         local cb = v[1]
-        if not cb:get(ns, 1) then
-            cb:add(ns, 1, 0/0) -- always advance the buffer/graphs
```

Comment:
What happened to this advancement? Is it part of the latency estimation logic now?